sysdiglabs/terraform-provider-sysdig
# feat(secure): Add Malware, Drift, ML and AWS ML policy resources

## Summary

- This PR adds support for Malware, Drift, ML and AWS ML policies.
- See the screenshots below for the **draft** TF resource design (the final design might deviate).

## Testing steps

1. Run `make install` in local `terraform-provider-sysdig` directory
2. Create a new directory for customer-side TF script
3. Create a new file called `main.tf` in that directory
4. Run `terraform init` (precede with `rm .terraform.lock.hcl` if you've changed the code), update the values if needed, and run `terraform apply`. Enter "yes" to confirm the changes. 
5. You should see a successful message

```terraform
// main.tf
terraform {
  required_providers {
    sysdig = {
      source  = "local/sysdiglabs/sysdig"
      version = "~> 1.0.0"
    }
  }
}

provider "sysdig" {
  sysdig_secure_url          = "https://secure-staging.sysdig.com"
  sysdig_secure_api_token    = "<TOKEN>"
  sysdig_secure_insecure_tls = true
}

resource "sysdig_secure_notification_channel_email" "email" {
  name       = "Test Email Channel 14"
  recipients = ["alexey.ovriakh@sysdig.com"]
}

data "sysdig_secure_malware_policy" "p" {
  name = "Test Malware Policy 9"
}

resource "sysdig_secure_malware_policy" "test_policy" {
  name        = "Test Malware Policy 14"
  description = "Test Malware Policy Description"
  enabled     = true
  severity    = 4

  rules {
    enabled = true
    description = "Test Malware Rule Description"
    tags        = ["tag1", "tag2"]

    details {
      use_managed_hashes = true

      additional_hashes {
        hash         = "304ef4cdda3463b24bf53f9cdd69ad3ecdab0842e7e70e2f3cfbb9f14e1c4ae6"
        hash_aliases = ["hash_name", "hash_name_2"]
      }

      ignore_hashes {
        hash         = "6ac3c336e4094835293a3fed8a4b5fedde1b5e2626d9838fed50693bba00af0e"
        hash_aliases = ["ignore_hashes", "ignore_hashes_2"]
      }
    }
  }
  
  actions {
    prevent_malware = true
    container = "stop"
    # capture
  }

  notification_channels = [sysdig_secure_notification_channel_email.email.id]
}

output "sysdig_secure_malware_policy_1" {
  value = sysdig_secure_malware_policy.test_policy
}

output "sysdig_secure_malware_policy_x" {
  value = data.sysdig_secure_malware_policy.p
}

```

## Draft design 
<img width="814" alt="Screenshot 2024-01-26 at 15 59 54" src="https://github.com/sysdiglabs/terraform-provider-sysdig/assets/142799039/762ecc5d-b0d1-43b2-8f26-5b3fcc37b4e2">
<img width="775" alt="Screenshot 2024-01-26 at 15 57 03" src="https://github.com/sysdiglabs/terraform-provider-sysdig/assets/142799039/1d7780f8-64ea-4f1a-b9e6-c7bb4bba0fc9">
<img width="832" alt="Screenshot 2024-01-26 at 15 56 54" src="https://github.com/sysdiglabs/terraform-provider-sysdig/assets/142799039/7e88e89c-a194-44c8-8d88-9da47af0a0e9">
<img width="849" alt="Screenshot 2024-01-26 at 15 56 45" src="https://github.com/sysdiglabs/terraform-provider-sysdig/assets/142799039/11366f44-465d-4598-b741-10c83f51f92e">
